File: android/okhttp_profiler_interceptor/src/main/java/com/gektor650/okhttp_profiler_interceptor/MessageType.java
package com.gektor650.okhttp_profiler_interceptor;

enum MessageType {
    INITIAL("INL"),
    REQUEST_HEADER("RQH"),
    REQUEST_BODY("RQB"),
    REQUEST_END("RQD"),
    RESPONSE_HEADER("RSH"),
    RESPONSE_BODY("RSB"),
    RESPONSE_END("RSD"),
    UNKNOWN("UNKNOWN");

    public final String name;

    MessageType(String name) {
        this.name = name;
    }
}

File: plugin/src/com/gektor650/MessageType.kt
package com.gektor650

enum class MessageType(val api: String) {
    REQUEST_URL("RQU"),
    REQUEST_METHOD("RQM"),
    REQUEST_HEADER("RQH"),
    REQUEST_BODY("RQB"),
    REQUEST_END("RQD"),
    RESPONSE_TIME("RST"),
    RESPONSE_STATUS("RSS"),
    RESPONSE_HEADER("RSH"),
    RESPONSE_BODY("RSB"),
    RESPONSE_END("RSD"),
    UNKNOWN("UNKNOWN");
    companion object {
        fun fromString(type: String): MessageType {
            for (value in values()) {
                if (type == value.api) {
                    return value
                }
            }
            return UNKNOWN
        }
    }
}